# Hypothesis Testing Exercise - Costomer+OrderForm data

Problem Statement :
TeleCall uses 4 centers around the globe to process customer order forms. They audit a certain % of the customer order forms. Any error in order form renders it defective and has to be reworked before processing. The manager wants to check whether the defective % varies by centre. Please analyze the data at 5% significance level and help the manager draw appropriate inferences


# Business Problem

Does the defective % varies significantly by centre ?

# Data Description

α == 0.05 (95% Confidence)

Y == Discrete
X == Discrete

Since there are more than 2 variable we will perform Chi-Square test

In [6]:
import pandas as pd
import numpy as np
from scipy import stats

In [7]:
ls

 Volume in drive C has no label.
 Volume Serial Number is C2AC-6915

 Directory of C:\Users\user\EXCELR\Assignments\Hypothesis testing

04/01/2022  08:35 PM    <DIR>          .
03/30/2022  10:44 AM    <DIR>          ..
04/01/2022  08:17 PM    <DIR>          .ipynb_checkpoints
03/30/2022  10:43 AM                87 BuyerRatio.csv
03/30/2022  10:43 AM            13,423 Costomer+OrderForm.csv
03/30/2022  10:43 AM               525 Cutlets.csv
03/30/2022  11:10 AM             6,898 Hypothesis Testing Exercise - Cutlets.mtw data.ipynb
03/30/2022  11:26 AM            10,275 Hypothesis Testing Exercise - LabTat data.ipynb
03/30/2022  10:43 AM            61,688 Hypothesis_Testing_Assignment.pptx
03/30/2022  10:43 AM             3,487 LabTAT.csv
04/01/2022  08:35 PM             2,309 Untitled.ipynb
               8 File(s)         98,692 bytes
               3 Dir(s)  171,797,200,896 bytes free


# Chi-Square Test

In [8]:
df=pd.read_csv("Costomer+OrderForm.csv")

In [9]:
df.head()

Phillippines   Indonesia       Malta       India
0   Error Free  Error Free   Defective  Error Free
1   Error Free  Error Free  Error Free   Defective
2   Error Free   Defective   Defective  Error Free
3   Error Free  Error Free  Error Free  Error Free
4   Error Free  Error Free   Defective  Error Free

H0 == The defective % does not varies by centre
H1 == The defective % does varies by centre

In [10]:
df.shape

(300, 4)

In [11]:
stats.chi2_contingency([df['Phillippines'].value_counts(), df['Indonesia'].value_counts(), df['Malta'].value_counts(), df['India'].value_counts()])



(3.8589606858203545,
 0.2771020991233144,
 3,
 array([[271.75,  28.25],
        [271.75,  28.25],
        [271.75,  28.25],
        [271.75,  28.25]]))

P value of Chi-Square test == 0.277 > α

Thus HO is accepted.


The defective % does not varies significantly by centres